sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -122,7 +127,10 @@ case object SinglePartition extends Partitioning {
     case _ => false
   }
 
-  override def keyExpressions: Seq[Expression] = Nil
+  override def guarantees(other: Partitioning): Boolean = other match {
```

Comment:
Shouldn't `SinglePartition` technically guarantee any partitioning which produces a single partition, such as `HashPartitioning` with a single partition?  I guess that hash partitioning with one partition shouldn't ever occur, though.